### DISCUSSION TITLE
Can we use optimistic updates with nested mutations?

### DISCUSSION THREAD

**Discussion by alexkirsz** [urql-graphql/urql#2325] [Q&A]
urql version & exchanges:

```json
    "@urql/core": "^2.4.2",
    "@urql/exchange-auth": "^0.1.7",
    "@urql/exchange-graphcache": "^4.3.6",
```

Let's say I have a `Mutation` type with some nested fields:

```gql
type Mutation = {
  posts: PostMutations
}

type PostMutations {
  createPost($post: InputPost!): Post!
}
```

How would I specify an optimistic update for `createPost` here? Per the doc, [the `optimistic` field of the graphcache exchange only appears to support top-level mutations](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#optimistic-updates). 

Cheers,
Alex

**Comment by thdxr** [comment]
Wondering about this as well
